applications/ContactStructuralMechanicsApplication/custom_utilities/contact_utilities.h

```diff
@@ -78,6 +91,95 @@ class ContactUtilities
     ///@name Operations
     ///@{
     
+    /**
+     * @brief This function computes the relative size of the mesh
+     * @param rThisModelPart The modelpart to compute
+     */
+
+    static inline double CalculateRelativeSizeMesh(ModelPart& rThisModelPart)
+    {
+        return CalculateMaxNodalH(rThisModelPart)/CalculateMinimalNodalH(rThisModelPart);
+    }
+
+    /**
+     * @brief This method computes the maximal nodal H
+     * @param rThisModelPart The modelpart to compute
+     */
+    static inline double CalculateMaxNodalH(ModelPart& rThisModelPart)
+    {
+        // We iterate over the nodes
+        NodesArrayType& nodes_array = rThisModelPart.Nodes();
+
+        // Creating a buffer for parallel vector fill
+        const int num_threads = OpenMPUtils::GetNumThreads();
+        std::vector<double> max_vector(num_threads, 0.0);
```

Comment:
Why don't you use a reduction here too?
Same what you are using in `CalculateMeanNodalH`
imo the better way and should simplify the code

Reply:
I didn't know it could be used for max and min, nice :dancer: